### PR TITLE
Automatically reload the server when it changes

### LINKED ISSUE #2426: `npm start` with autoreload
:clipboard: **Description**

In TUTORIAL.md I need to start and stop `npm` server every time to load new changes? That's cumbersome. Is there any autoreload mechanism for /services/?

### PR DESCRIPTION
**Pull request by paulmelnikow**
While working on #2428 I found myself wanting to reload the server frequently. This is working great and reducing my iteration time significantly. I should have tackled this way sooner! 🙊 

I’ve left `verbose` on which seems useful, at least in the short term while we’re tuning the configuration.

Close #2426

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -103,12 +103,14 @@
     "defs": "node scripts/export-service-definitions-cli.js > service-definitions.yml",
     "build": "run-s defs features && gatsby build",
     "heroku-postbuild": "run-s --silent build",
-    "start:server": "cross-env NODE_CONFIG_ENV=development node server 8080",
-    "now-start": "node server",
+    "start:server:prod": "node server",
+    "now-start": "npm run start:server:prod",
```
Comment (calebcartwright) [inline package.json]: Where do we use this script?

File: doc/TUTORIAL.md [hunk 2]
```diff
@@ -152,10 +152,9 @@ The process of turning this object into an image is handled automatically by the
 To try out this example badge:
 
 1. Copy and paste this code into a new file in `/services/example/example.service.js`
-2. Quit the running server with `Control+C`.
-3. Start the server again.
-   `npm start`
-4. Visit the badge at <http://localhost:8080/example/foo.svg>.
+2. The server should restart on its own. (If it doesn't for some reason, quit
```
Comment (calebcartwright) [inline doc/TUTORIAL.md]: 👍 I like the auto-reload call out here. I feel like we should also call out that behavior elsewhere in other docs (so folks know they don't need to manually restart on changes). 

I was hoping to propose some specific suggestions but can't get the wording right.

File: package.json (hunk 1 above)
Comment (paulmelnikow) [reply package.json]: It’s for running Shields on Zeit Now.

File: doc/TUTORIAL.md (hunk 2 above)
Comment (paulmelnikow) [reply doc/TUTORIAL.md]: Could probably add it in the readme in the spot I updated!

Though FWIW the console is pretty chatty so it’s pretty easy to notice!

File: package.json [hunk 3]
```diff
@@ -103,12 +103,14 @@
...
+    "start:server:e2e-on-build": "node server 8080",
+    "start:server": "cross-env NODE_CONFIG_ENV=development nodemon server 8080",
     "prestart": "run-s --silent depcheck defs features",
     "start": "concurrently --names server,frontend \"npm run start:server\" \"cross-env GATSBY_BASE_URL=http://localhost:8080 gatsby develop --port 3000\"",
     "e2e": "start-server-and-test start http://localhost:3000 test:e2e",
```
Comment (platan) [inline package.json]: In `npm run e2e` I got:
```
[server] [nodemon] ignoring: package.json **/*.spec.js **/*.tester.js **/*.integration.js ./frontend/**/* ./public/**/*
[server] [nodemon] watching: *.*
[server] [nodemon] watching extensions: js
...
  Main page
[server] [nodemon] files triggering change check: cypress/videos/main-page.spec.js.mp4
[server] [nodemon] matched rule: **/*.*
[server] [nodemon] changes after filters (before/after): 1/0
[server] [nodemon] files triggering change check: cypress/videos/main-page.spec.js.mp4
[server] [nodemon] matched rule: **/*.*
[server] [nodemon] changes after filters (before/after): 1/0
```
So I added `"**/*.js.*",` to `nodemonConfig.ignore`
In another run I got:
```
  - Started processing:   Compressing to 32 CRF
[server] [nodemon] files triggering change check: cypress/videos/main-page.spec.js-compressed.mp4
[server] [nodemon] matched rule: **/*.*
[server] [nodemon] changes after filters (before/after): 1/0
[server] [nodemon] files triggering change check: cypress/videos/main-page.spec.js-compressed.mp4
[server] [nodemon] matched rule: **/*.*
[server] [nodemon] changes after filters (before/after): 1/0
[server] [nodemon] files triggering change check: cypress/videos/main-page.spec.js-compressed.mp4
[server] [nodemon] matched rule: **/*.*
```
So I added `"**/*.js-*.*",` and after this I have no info about change check. Changes in mp4 does not restart server but it's strange for me that `main-page.spec.js.mp4` and `main-page.spec.js-compressed.mp4` trigger checks.

Comment (paulmelnikow) [reply package.json]: Would ignoring `cypress/` solve this?

Comment (platan) [reply package.json]: Sounds good to me :-)

File: doc/TUTORIAL.md (hunk 2 above)
Comment (paulmelnikow) [reply doc/TUTORIAL.md]: Good shout. I've added a few sentences in the readme.

File: README.md [hunk 4]
```diff
@@ -98,9 +98,15 @@ You can read a [tutorial on how to add a badge][tutorial].
...
-4. Run `npm start` to start the server.
+4. Run `npm start` to start the badge server and the frontend dev server.
 5. Open `http://localhost:3000/` to view the frontend.
 
+When server source files change, the badge server should automatically restart
+itself (using [nodemon][]). When the frontend files change, the frontend dev
+server (`gatsby dev`) should also automatically reload. However the badge
+definitions are built only before the server first starts. To regenerate those,
+either run `npm run defs` or manually restart the server.
+
```
Comment (calebcartwright) [inline README.md]: 👍

**Comment by paulmelnikow** [comment]
Thanks for reviewing and testing!

**Pull request merged**